python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -497,12 +490,13 @@ def _logs_thread(
                         > WAIT_FOR_LOGS_TIMEOUT
                     ):
                         for key, log_reader in unstarted_log_readers.items():
-                            warnings.warn(
-                                f"[pipes] Attempted to read log for reader {key}:{log_reader.name} but log was"
-                                f" still not written {WAIT_FOR_LOGS_TIMEOUT} seconds after session close. Abandoning reader {key}."
-                            )
-
-                        return
+                            if key not in failed_log_readers:
+                                failed_log_readers.add(key)
```

Comment:
Confused about `failed_log_readers` here. Seems like it's not doing anything? If there is only one loop over `unstarted_log_readers` keys, how could we ever loop over the same key twice (and so why would we need to track it in `failed_log_readers`)?

Reply:
was an old variable leaked from a different PR. Removed it.